src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -385,30 +388,33 @@ public IEnumerable<AzureTenant> ListTenants()
             return ListAccountTenants(_profile.Context.Account, _profile.Context.Environment, null, ShowDialog.Never);
         }
 
-        public IEnumerable<AzureSubscription> ListSubscriptions(string tenant, ref string listNextLink)
-        {
-            return ListSubscriptionsForTenant(
-                _profile.Context.Account,
-                _profile.Context.Environment,
-                null,
-                ShowDialog.Never,
-                tenant,
-                ref listNextLink);
-        }
-
-        public IEnumerable<AzureSubscription> ListSubscriptions()
+        public IEnumerable<AzureSubscription> ListSubscriptions(string tenantIdOrDomain = "")
         {
             string listNextLink = null;
 
+            var tenants = new List<AzureTenant>();
+
+            if (string.IsNullOrWhiteSpace(tenantIdOrDomain))
+            {
+                tenants.AddRange(ListTenants());
+            }
+            else
+            {
+                tenants.Add(CreateTenant(tenantIdOrDomain));
+            }
+
             List<AzureSubscription> subscriptions = new List<AzureSubscription>();
-            foreach (var tenant in ListTenants())
+            foreach (var tenant in tenants)
             {
                 try
```

Comment:
@markcowl no longer needed since `RMProfileClient` no longer has knowledge of nextLink